2018/003-offline-chart-repository.md

```diff
@@ -0,0 +1,441 @@
+that, once installed, will create the following files:
+
+  * `/usr/share/suse-kubernetes-charts/ingress-1.1.0.tgz`
+  * `/usr/share/suse-kubernetes-charts/ingress-2.0.0.tgz`
+  * `/usr/share/suse-kubernetes-charts/monitoring-2.0.1.tgz`
+  * `/usr/share/suse-kubernetes-charts/dashboard-3.0.0.tgz`
+
+The SUSE Cloud Application Platform team will provide a set of rpms that,
+once installed, will create the following files:
+
+  * `/usr/share/suse-kubernetes-charts/cf-2.0.2.tgz`
+  * `/usr/share/suse-kubernetes-charts/cf-2.10.1.tgz`
+  * `/usr/share/suse-kubernetes-charts/cf-2.6.11.tgz`
+  * `/usr/share/suse-kubernetes-charts/cf-2.7.0.tgz`
+  * `/usr/share/suse-kubernetes-charts/cf-2.8.0.tgz`
+  * `/usr/share/suse-kubernetes-charts/cf-usb-sidecar-mysql-1.0.1.tgz`
+  * `/usr/share/suse-kubernetes-charts/cf-usb-sidecar-postgres-1.0.1.tgz`
+  * `/usr/share/suse-kubernetes-charts/console-0.9.7.tgz`
+  * `/usr/share/suse-kubernetes-charts/console-0.9.8.tgz`
+  * `/usr/share/suse-kubernetes-charts/console-0.9.9.tgz`
+  * `/usr/share/suse-kubernetes-charts/console-1.0.0.tgz`
+  * `/usr/share/suse-kubernetes-charts/console-1.0.2.tgz`
+  * `/usr/share/suse-kubernetes-charts/console-1.1.0.tgz`
+  * `/usr/share/suse-kubernetes-charts/uaa-2.0.2.tgz`
+  * `/usr/share/suse-kubernetes-charts/uaa-2.10.1.tgz`
+  * `/usr/share/suse-kubernetes-charts/uaa-2.6.11.tgz`
+  * `/usr/share/suse-kubernetes-charts/uaa-2.7.0.tgz`
+  * `/usr/share/suse-kubernetes-charts/uaa-2.8.0.tgz`
+
+The same thing will be done by the other teams at SUSE who want to ship containerized
+workloads managed by helm.
+
+#### Building the offline chart repository image
+
+The `registry.suse.com/suse/kubernetes-charts` will be built inside of the
+Open Build Service (obviously inside of IBS).
+
+The build service project will consists of:
+
+  * RPM packages of helm charts.
+  * RPM defining patterns of helm charts.
+  * Definition of the `registry.suse.com/suse/kubernetes-charts` container image.
+
+The next sections will provide details about these resources.
+
+##### RPM packages definitions
+
+Each helm chart should have its own RPM package. For example we would have a
+`cf-chart`, `console-chart`, `uaa-chart`, …
+
+We must be able to ship multiple versions of the same helm chart (eg: `cf-2.0.2.tgz` and `cf-2.10.1.tgz`). To achieve that each RPM will install one `.tgz` file per version of the chart.
+
+For example, the build service package for the `cf-chart` RPM would consist of the following files:
+
+  * `cf-chart.spec`
+  * `cf-chart.changelog`
+  * `cf-2.0.2.tgz`
+  * `cf-2.10.1.tgz`
+
+Once installed the RPM would provide the following files:
+
+  * `/usr/share/suse/kubernetes-charts/cf-2.0.2.tgz`
+  * `/usr/share/suse/kubernetes-charts/cf-2.10.1.tgz`
+
+In the future each chart RPM should also provide the `.tgz.prov` provenance files. However this RFC doesn't want to cover the problem of creating these signed files (currently they are missing also from *”kubernetes-charts.suse.com”*).
```

Comment:
** Agreed with others that this is something we should address up front. It is not the most straight forward process but it is just GPG under the hood.